src/core/Akka/Serialization/Serialization.cs

```diff
@@ -145,14 +146,15 @@ private Serializer GetSerializerByName(string name)
         public ActorSystem System { get; }
 
         /// <summary>
-        /// TBD
+        /// Adds the serializer to the internal state of the serialization subsystem
         /// </summary>
-        /// <param name="serializer">TBD</param>
-        /// <returns>TBD</returns>
+        /// <param name="name">Configuration name of the serializer</param>
+        /// <param name="serializer">Serializer instance</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void AddSerializer(Serializer serializer)
+        public void AddSerializer(string name, Serializer serializer)
         {
             _serializers.Add(serializer.Identifier, serializer);
+            _serializerIdByName.Add(name, serializer.Identifier);
```

Comment:
Ok so technically, the Identifier is retrieved twice. But since Hocon caching is a thing. I found this to be more expressive then first storing the id in a temp var and using that.

Reply:
I'd take a look at my previous comment and consider just maintaining two `<TKey, Serializer>` dictionaries - the ID-based one and the name based one, rather than go name --> id --> serializer you can just go name --> serializer. Does that make sense?